pyhmmer/easel.pyx

```diff
@@ -5218,16 +5218,18 @@ cdef class DigitalSequence(Sequence):
             raise AllocationError("ESL_DSQ", sizeof(ESL_DSQ), aalen + 2)
         elif status != libeasel.eslOK:
             raise UnexpectedError(status, "esl_sq_Grow")
-                
-        # translate sequence 
+
+        # translate & rename sequence
         with nogil:
             genetic_code._translate(&self._sq.dsq[1], ntlen, &protein._sq.dsq[1], aalen)
             protein._sq.dsq[0] = protein._sq.dsq[aalen+1] = libeasel.eslDSQ_SENTINEL
+            status = libeasel.sq.esl_sq_SetName(protein._sq, self._sq.name)
```

Comment:
Just do `protein.name` = `self.name` here, as this will take care of raising exceptions, which is not the case here (unless you manually check the content of status`).

Reply:
Or like you did later, create the `protein` object directly with the name attribute set.